xrootd/client/client_test.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build xrootd_test_with_server
+
+package client // import "go-hep.org/x/hep/xrootd/client"
+
+import (
+	"context"
+	"testing"
+)
+
+func TestNewClient(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	client, err := NewClient(ctx, *Addr)
```

Comment:
ok.
instead of this `Addr` flag value I would rather prefer something like this:

```go
var testClientAddrs = []string{"0.0.0.0:9001"}

func TestNewClient(t *testing.T) {
    for _, addr := range testClientAddrs {
        t.Run(addr, func(t *testing.T) {
            testNewClient(t, addr)
        })
    }
}

func testNewClient(t *testing.T, addr string) { ... }
```

then, in some other `_test.go` file, perhaps protected with a specific build tag:
```go
// +build cxx-xrootd

package client

func init() {
    // add a C++ XRootD test server hosted in CC-Lyon.
    testClientAddrs = append(testClientAddrs, "ccxrootdgotest.in2p3.fr:9001")
}
```

what do you think ?

the advantage is that it's more explicit and the expected instructions on how to run the test(s) are kind of "documented in the code" (instead of some arcane way to run the test.)